text/0018-publish-update.md

```diff
@@ -0,0 +1,180 @@
+
+Speaking of updates, I don't have any clear mechanism for re-fetching datasets that's been updated since I "added" it. We need a quick and easy way to syncronize my remote copy of a dataset with the lastest version after it's been published.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This RFC introduces two new concepts to Qri that will come with new commands and, will modify some existing behaviors. Publish and Update will work together to allow users to explicitly opt-in to sync points
+
+Because this process will be run frequently, I'd like to make the common use patterns as simple as possible, to the point that `qri save` should come with a `publish` flag that combines saving an update & publishing it into one step.
+
+## Publish
+All datasets start as "unlisted", and the user must `publish` them to make visible to other users. Peers calling `list_datasets` now only see datasets you've published, at version that's been most-recently published. If I create a dataset, then publish, and save changes to that dataset, I must re-publish the new changes for others to see the update.
+
+Whenever I choose, I may "unpublish" a dataset, which will remove it entirely from the list returned by `list_datasets`. It is true that others may still have a copy of my dataset. At present there's not much I can do about it.
+
+#### Publish will integrate with Registries
+Running `publish` with a configured registry should POST a dataset _summary_ to the registry. We haven't defined dataset previews, summaries, and documents yet (a story for another RFC), but once it's done publish will both list the dataset for p2p discovery and push the dataset to the registry.
+
+## Update
+Update brings your data to the latest known version of a dataset. Update comes in two flavors:
+* Updating Someone else's dataset
+* Updating A Dataset You Own
+Both types of update will be accessed via the same command like and API interfaces, they only differ based on the dataset reference in question.
+
+#### Updating a peer's dataset
+Running update on a peer dataset checks to see if the peer is online and calls `get_dataset`, if the version is different, it'll grab that dataset. If the peer in question is offline we can't check for updates. Once publish pushes to the registry we can check the registry for updates.
+
+Later on we'll want to do proactive checks for updates by submitting a list of dataset refs we'd like to grab updates for to peers and registries.
+
+#### Updating a dataset you own
+running update on a dataset in your namespace will check if a transform script is specified, If so, it'll re-execute the transform & check for changes. If changes exist it'll create a commit.
```

Comment:
This makes it possible to define a transform script that breaks updates in a non-obvious way.

For example:

```
Dataset with body:
{"name": "Dustin"}
```
```
def transform(ds, context):
  prev = ds.load_previous_version()
  ds.save({"name": prev["name"] + "g"})
```

Every time this runs `update`, it'll append an additional "g", therefore it always has changes. This is true of any nearly any transform script that modifies the previous version somehow. This makes it hard to treat update like a "regularly update my sets" kind of cronjob-like task.

Perhaps instead we want `update` to only check for changes to external sources of data? Or perhaps there's a way to take advantage of our pipeline / special functions, and only follow through with changes that are generated outside of the `transform` function?

Reply:
"Updating a dataset in your namespace"